### ISSUE TITLE
[All] Please compress assets, your games could be several times smaller

### ISSUE THREAD

**Issue by ghost** [ValveSoftware/Source-1-Games#1790]
Yes, yes, disk space is cheap, but uncompressed files are slow to read, and we all hate excessive loading times. More importantly, though, smaller files use up less bandwidth.

Your bandwidth.

So go compress your games and save yourselves some money.

Most files can just be gzipped (or xz'd). The typical compression ratio here is 2x, although it's lower for audio, higher for maps, and even higher for raw text.

For audio, use opus with a bitrate between 48 and 96 kbps. That's over 7x smaller than the uncompressed wavs (705 kbps) currently being used.

Unfortunately, there is no One True Format for textures, so just gzipping the VTF files is a valid choice here. However, here are some tips for better compressing textures:
- PNG is, of course, the most commonly used lossless image format
- Lossy formats available are JPEG, JPEG-XR, JPEG-2000, or WebP.
- For a more ambitious solution, take advantage of similarity between related textures and use a video codec to compress them as an image sequence.
- The current industry standard is h264, although vp8 is a lower-quality alternative that doesn't have any patent issues.

**Comment by MRebhan** [comment]
@FirstRoboticsGuy999 that is not true. MP3 files compared to WAV definitely loses some quality, same as compressed vs uncompressed PNG. But algorithms like XZ or ZIP or whatever do _not_.
